artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -1153,6 +1161,31 @@ public boolean isFull() {
       return maxSize > 0 && getAddressSize() > maxSize || pagingManager.isGlobalFull();
    }
 
+
+   @Override
+   public int getAddressLimitPercent() {
+      final long currentUsage = getAddressSize();
+      if (currentUsage != 0) {
+         if (maxSize > 0) {
+            return Math.toIntExact((currentUsage / maxSize) * 100);
```

Comment:
Dividing currentUsage by maxSize should always return 0

Reply:
Or 1, if currentUsage is the same (or bigger but < double).

Phrased another way: unit tests missing with 0% !=  x != 100% checks also?

(the currentUsage != 0 could probably go away too)

Reply:
that does need work and tests, on it, thanks!